packages/Kokkos/test/tstKokkosHelpers.cpp

```diff
@@ -0,0 +1,232 @@
+//---------------------------------------------------------------------------//
+/*
+  Copyright (c) 2012, Stuart R. Slattery
+  All rights reserved.
+
+  Redistribution and use in source and binary forms, with or without
+  modification, are permitted provided that the following conditions are
+  met:
+
+  *: Redistributions of source code must retain the above copyright
+  notice, this list of conditions and the following disclaimer.
+
+  *: Redistributions in binary form must reproduce the above copyright
+  notice, this list of conditions and the following disclaimer in the
+  documentation and/or other materials provided with the distribution.
+
+  *: Neither the name of the University of Wisconsin - Madison nor the
+  names of its contributors may be used to endorse or promote products
+  derived from this software without specific prior written permission.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+  HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+//---------------------------------------------------------------------------//
+/*!
+ * \file   tstKokkosHelpers.cpp
+ * \author Stuart Slattery
+ * \brief  Kokkos helpers unit tests.
+ */
+//---------------------------------------------------------------------------//
+
+#include "DTK_DBC.hpp"
+#include "DTK_KokkosHelpers.hpp"
+
+#include <Kokkos_Core.hpp>
+
+#include <Teuchos_UnitTestHarness.hpp>
+
+#include <limits>
+#include <type_traits>
+
+//---------------------------------------------------------------------------//
+// TEST FUNCTORS
+//---------------------------------------------------------------------------//
+// Algorithm tags.
+struct MinValue
+{
+};
+struct MaxValue
+{
+};
+struct LimitsMax
+{
+};
+struct LimitsEpsilon
+{
+};
+
+//---------------------------------------------------------------------------//
+// Fill function - specialize this for different fill operations.
+template <class View, class Algorithm>
+KOKKOS_INLINE_FUNCTION typename View::traits::value_type
+fillFunction( Algorithm alg );
+
+// Fill with the max of 0 and 1
+template <class View>
+KOKKOS_INLINE_FUNCTION typename View::traits::value_type
+fillFunction( MaxValue alg )
```

Comment:
Either don't put `alg` or use `(void) alg`. Otherwise, you will get a warning about `alg` not being used.